include/deal.II/base/function.h

```diff
@@ -347,135 +347,158 @@ class Function : public FunctionTime<Number>,
 };
 
 
-
-/**
- * Provide a function which always returns zero. Obviously, also the derivates
- * of this function are zero. Also, it returns zero on all components in case
- * the function is not a scalar one, which can be obtained by passing the
- * constructor the appropriate number of components.
- *
- * This function is of use when you want to implement homogeneous boundary
- * conditions, or zero initial conditions.
- *
- * @ingroup functions
- * @author Wolfgang Bangerth, 1998, 1999
- */
-template <int dim, typename Number=double>
-class ZeroFunction : public Function<dim, Number>
+namespace Functions
 {
-public:
-  /**
-   * Constructor. The number of components is preset to one.
-   */
-  ZeroFunction (const unsigned int n_components = 1);
+
 
   /**
-   * Virtual destructor; absolutely necessary in this case.
+   * Provide a function which always returns zero. Obviously, also the derivatives
+   * of this function are zero. Also, it returns zero on all components in case
+   * the function is not a scalar one, which can be obtained by passing the
+   * constructor the appropriate number of components.
    *
+   * This function is of use when you want to implement homogeneous boundary
+   * conditions, or zero initial conditions.
+   *
+   * @ingroup functions
+   * @author Wolfgang Bangerth, 1998, 1999
    */
-  virtual ~ZeroFunction ();
+  template <int dim, typename Number=double>
+  class ZeroFunction : public Function<dim, Number>
+  {
+  public:
+    /**
+     * Constructor. The number of components is preset to one.
+     */
+    ZeroFunction (const unsigned int n_components = 1);
 
-  virtual Number value (const Point<dim>   &p,
-                        const unsigned int  component) const;
+    /**
+     * Virtual destructor; absolutely necessary in this case.
+     *
```

Comment:
no need to add the empty line.

I think I would also rewrite the documentation to just say "Destructor". The rationale made sense to us 15 years ago, but I think it does no longer.